Documentation/project-docs/using-nightly-builds.md

```diff
@@ -0,0 +1,244 @@
+
+```csharp
+using System;
+using System.Net;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        WebUtility.HtmlDecode("&amp;", Console.Out);
+        Console.WriteLine();
+        Console.WriteLine("Hello World!");
+    }
+}
+```
+
+Run the bits:
+
+```
+$ dotnet run
+```
+
+Rinse and repeat!
+
+## Advanced Scenario - Using a nightly build of Microsoft.NETCore.App
+
+If you use the above instructions, your application will run against the same .NET Core 2.0 runtime that comes with the latest SDK. That works fine to get up and running quickly. However, there are times when you need to use a nightly build of Microsoft.NETCore.App which hasn't made its way into the SDK yet. To enable this, there are two options you can take.
+
+### Option 1: Framework-dependent
+
+This is the default case for applications - running against an installed .NET Core runtime.
+
+0. You still need to install the prerequisite .NET Core SDK from above.
+1. Also, install the specific .NET Core runtime you require:
+    - https://github.com/dotnet/core-setup#daily-builds
+    - Remember the version number you picked, you'll need it below.
+2. Modify your `.csproj` to reference the nightly build of Microsoft.NETCore.App.
+
+```xml
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <RuntimeFrameworkVersion>2.0.0-beta-xyz-00</RuntimeFrameworkVersion> <!-- pick nightly build -->
+  </PropertyGroup>
+```
+
+```
+$ dotnet restore
+$ dotnet run
+```
+
+### Option 2: Self-contained
+
+In this case, the .NET Core runtime will be published alongside your application.
+
+0. You still need to install the prerequisite .NET Core SDK from above.
+1. Modify your `.csproj` to reference the nightly build of Microsoft.NETCore.App *and*
+make it self-contained.
+
+```xml
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <RuntimeFrameworkVersion>2.0.0-beta-xyz-00</RuntimeFrameworkVersion> <!-- pick nightly build -->
+    <RuntimeIdentifier>win7-x64</RuntimeIdentifier> <!-- make self-contained -->
+  </PropertyGroup>
+```
+
+```
+$ dotnet restore
+$ dotnet publish
+$ bin\Debug\netcoreapp2.0\win7-x64\publish\App.exe
+```
+
+Note #1: There is a [bug](https://github.com/dotnet/sdk/issues/791) with `dotnet run` and self-contained applications. If you `dotnet run` and see an error `The library 'hostpolicy.dll' is required to execute the application...`, you've hit this bug.
+
+Note #2: On non-Windows platforms, self-contained applications aren't runnable by default. You will see an error "Permission denied" when running the application. This is because of a [breaking change in the .NET Core runtime](https://github.com/dotnet/corefx/issues/15516) between 1.0 and 2.0. Either this breaking change needs to be fixed, or [NuGet will have to workaround the change](https://github.com/NuGet/Home/issues/4424).
```

Comment:
The issues mentioned in both notes above are now closed. Does that mean the two notes no longer apply and should be deleted?

Reply:
Yeah, I don't think these are relevant anymore.